### PR TITLE
chore: Restore self-deleting messages SQSERVICES-1059

### PR DESCRIPTION
**Pull request by KaterinaWire**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/SQSERVICES-1059" title="SQSERVICES-1059" target="_blank">TaskSQSERVICES-1059</a>  Restore self-deleting messages
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?
Restore self-deleting messages

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [x] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: Patches/InvalidConversationRemoval.swift [hunk 1]
```diff
@@ -26,6 +28,7 @@ enum InvalidConversationRemoval {
         do {
             try moc.batchDeleteEntities(named: ZMConversation.entityName(), matching: NSPredicate(format: "\(ZMConversationConversationTypeKey) == \(ZMConversationType.invalid.rawValue)"))
         } catch {
+            zmLog.safePublic("Failed to batch delete entities: \(error.localizedDescription)")
```
Comment (johnxnguyen) [inline Patches/InvalidConversationRemoval.swift:31]: Changes in this file should be reverted too.

File: Patches/PersistedDataPatches.swift [hunk 2]
```diff
@@ -36,60 +36,24 @@ public struct PersistedDataPatch {
...
-            return zmLog.safePublic("Can't retrieve bundle for data model, skipping patches..")
-        }
-
-        guard let currentVersion = bundle.infoDictionary!["CFBundleShortVersionString"] as? String else {
-            return zmLog.safePublic("Can't retrieve CFBundleShortVersionString for data model, skipping patches..")
-        }
-
-        zmLog.safePublic("current version is: \(currentVersion)")
-        
+        let currentVersion = "278.0.0"
```
Comment (johnxnguyen) [inline Patches/PersistedDataPatches.swift:42]: I think a comment is needed here just to explain the situation, something like "Normally we fetch the short version from the framework bundle, but it's not working so we hardcode the version.

File: Resources/Configurations/version.xcconfig [hunk 3]
```diff
@@ -1,2 +1,2 @@
-MAJOR_VERSION = 277
-CURRENT_PROJECT_VERSION = 277.0.3
+MAJOR_VERSION = 272
+CURRENT_PROJECT_VERSION = 272.0.2
```
Comment (johnxnguyen) [inline Resources/Configurations/version.xcconfig:2]: You can also bump these, even though they're not being used.

**Review by johnxnguyen — CHANGES_REQUESTED** [review]

File: Resources/Configurations/version.xcconfig (hunk 3 above)
Comment (KaterinaWire) [reply Resources/Configurations/version.xcconfig]: Fixed

File: Patches/PersistedDataPatches.swift (hunk 2 above)
Comment (KaterinaWire) [reply Patches/PersistedDataPatches.swift:43]: Added comment

**Review by typfel — APPROVED** [review]

File: Patches/InvalidConversationRemoval.swift (hunk 1 above)
Comment (KaterinaWire) [reply Patches/InvalidConversationRemoval.swift]: Done

**Review by johnxnguyen — APPROVED** [review]

**Comment by KaterinaWire** [comment]
release major

**Pull request merged**